src/components/Translate/TranslateForm.vue

```diff
@@ -0,0 +1,240 @@
+	},
+
+	computed: {
+		outputEqualsInput() {
+			if (typeof this.inputs?.input === 'string' && typeof this.outputs?.output === 'string') {
+				return this.hasInitialOutput && this.outputs.output?.trim() === this.inputs.input?.trim()
+			}
+			return false
+		},
+		hasInitialOutput() {
+			return !!this.outputs?.output?.trim()
+		},
+	},
+
+	watch: {
+		translateTaskType() {
+			console.debug('[assistant] watch translateTaskType', this.translateTaskType, this.translateTaskTypeId)
+			this.setDefaultValues(true)
+		},
+	},
+
+	mounted() {
+		console.debug('[assistant] mounted TranslateForm', this.translateTaskId, this.translateTaskType)
+		// don't set the default values if there is a loaded task (initial or from history)
+		if (this.translateTaskId === null) {
+			this.setDefaultValues(false)
+		}
+
+		this.autosize()
+	},
+
+	methods: {
+		autosize() {
+			const inputField = document.getElementById('input-input')
+			const outputField = document.getElementById('output-input')
+			if (inputField && outputField) {
+				inputField.style.overflowY = 'hidden'
+				inputField.style.height = 'auto'
+				const height = inputField.scrollHeight + 10
+				inputField.style.height = height + 'px'
+				outputField.style.height = height + 'px'
+				inputField.style.overflowY = 'auto'
+			}
+		},
+		onValueChange(key, value) {
+			if (key === 'input') {
+				this.autosize()
+			}
+			const newValues = {
+				...this.inputs,
+				[key]: value,
+			}
+			console.debug('[assistant] translate field value change', newValues)
+			this.$emit('update:inputs', newValues)
+		},
+		onOutputValueChange(key, value) {
+			const newValues = {
+				...this.outputs,
+				[key]: value,
+			}
+			console.debug('[assistant] translate output field value change', newValues)
+			this.$emit('update:outputs', newValues)
+		},
+		setDefaultValues(clear = true) {
+			console.debug('[assistant] set translate default values', this.translateTaskType?.inputShapeDefaults, this.translateTaskType?.optionalInputShapeDefaults)
+			const inputs = clear
+				? {}
+				: {
+					...this.inputs,
+				}
+			// set default values
+			if (this.translateTaskType.inputShapeDefaults) {
+				Object.keys(this.translateTaskType.inputShapeDefaults).forEach(key => {
+					if (this.translateTaskType.inputShapeDefaults[key]) {
+						inputs[key] = this.translateTaskType.inputShapeDefaults[key]
+					}
+				})
+			}
+			if (this.translateTaskType.optionalInputShapeDefaults) {
+				Object.keys(this.translateTaskType.optionalInputShapeDefaults).forEach(key => {
+					if (this.translateTaskType.optionalInputShapeDefaults[key]) {
+						inputs[key] = this.translateTaskType.optionalInputShapeDefaults[key]
+					}
+				})
+			}
+			this.$emit('update:inputs', inputs)
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.assistant-inputs {
+	margin-bottom: 1rem;
+
+	.wrapper {
+		display: grid;
+		grid-template-columns: repeat(2, minmax(300px, 1fr));
+		grid-column-gap: 16px;
+
+		.col {
+			grid-row: 1/2;
+		}
+	}
```

Comment:
```suggestion
	.wrapper {
		display: flex;
		gap: 16px;

		.col {
			width: 100%;
		}
	}
```
Flex boxes are simpler to handle IMO.

Reply:
Good idea, maybe the translate modal in the Text app could use the same boxes too?